examples/simple/src/posts/PostEdit.tsx

```diff
@@ -1,253 +1,240 @@
+    onAddChoice(choice);
+    onCreate(choice);
+    setValue("");
+    return false;
+  };
+  return (
+    <Dialog open onClose={onCancel}>
+      <form onSubmit={handleSubmit}>
+        <DialogContent>
+          <MuiTextField
+            label="New Category"
+            value={value}
+            onChange={(event) => setValue(event.target.value)}
+            autoFocus
+          />
+        </DialogContent>
+        <DialogActions>
+          <Button type="submit">Save</Button>
+          <Button onClick={onCancel}>Cancel</Button>
+        </DialogActions>
+      </form>
+    </Dialog>
+  );
 };
 
 const EditActions = ({ basePath, data, hasShow }: EditActionsProps) => (
-    <TopToolbar>
-        <CloneButton
-            className="button-clone"
-            basePath={basePath}
-            record={data}
-        />
-        {hasShow && <ShowButton basePath={basePath} record={data} />}
-    </TopToolbar>
+  <TopToolbar>
+    <CloneButton className="button-clone" basePath={basePath} record={data} />
+    {hasShow && <ShowButton basePath={basePath} record={data} />}
+  </TopToolbar>
 );
 
 const SanitizedBox = ({
-    fullWidth,
-    basePath,
-    ...props
+  fullWidth,
+  basePath,
+  ...props
 }: BoxProps & { fullWidth?: boolean; basePath?: string }) => <Box {...props} />;
 
 const categories = [
-    { name: 'Tech', id: 'tech' },
-    { name: 'Lifestyle', id: 'lifestyle' },
+  { name: "Tech", id: "tech" },
+  { name: "Lifestyle", id: "lifestyle" }
 ];
 
 const PostEdit = ({ permissions, ...props }) => {
-    return (
-        <Edit title={<PostTitle />} actions={<EditActions />} {...props}>
-            <TabbedForm
-                initialValues={{ average_note: 0 }}
-                warnWhenUnsavedChanges
-            >
-                <FormTab label="post.form.summary">
-                    <SanitizedBox
-                        display="flex"
-                        flexDirection="column"
-                        width="100%"
-                        justifyContent="space-between"
-                        fullWidth
-                    >
-                        <TextInput disabled source="id" />
-                        <TextInput
-                            source="title"
-                            validate={required()}
-                            resettable
-                        />
-                    </SanitizedBox>
-                    <TextInput
-                        multiline={true}
-                        fullWidth={true}
-                        source="teaser"
-                        validate={required()}
-                        resettable
-                    />
-                    <CheckboxGroupInput
-                        source="notifications"
+  console.log("PostEdit", { props });
+  return (
+    <Edit title={<PostTitle />} actions={<EditActions />} {...props}>
+      <TabbedForm
+        initialValues={_.merge({}, { average_note: 0, pictures: [] })}
```

Comment:
I forgot to say that I also I added this part : a global object for the initialValues. Since the ArrayInput field prevents us from specifying here the shape of the `pictures` field, I had to also add `initialValue` at the field level. This global `initialValues` may be important for the bug.